### PR TITLE
Rumble extractor improvements

### PR DESCRIPTION
**Pull request by Benau**
The rumble channel descriptions seem to be only separated by `<p>` as far as I know.

### CODE REVIEW

**Comment by evermind-zz** [comment]
@Benau thank you for your contributions. I'll look into them this weekend.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/rumble/RumbleParsingHelper.java [hunk 1]
```diff
@@ -218,4 +218,18 @@ public static synchronized Map<String, List<String>> getMinimalHeaders() {
...
         return HEADERS;
     }
+
+    public static String getEmbedVideoId(final String rb) {
+        final String VALID_URL = "https?://(?:www\\.)?rumble\\.com/embed/(?:[0-9a-z]+\\.)?(?<id>[0-9a-z]+)";
+        final String EMBED_REGEX = "(?:<(?:script|iframe)[^>]+\\bsrc=|[\"']embedUrl[\"']\\s*:\\s*)[\"'](?<url>" + VALID_URL + ")";
+        Pattern pattern = Pattern.compile(EMBED_REGEX);
+        Matcher matcher = pattern.matcher(rb);
+        if (matcher.find()) {
+            String url = matcher.group("url");
```
Comment (evermind-zz) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/rumble/RumbleParsingHelper.java]: I like the group naming feature but as this feature is not available on older android versions (and desugaring does not support it) could you please use the matching group number instead?

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/rumble/RumbleParsingHelper.java [hunk 2]
```diff
@@ -218,4 +218,18 @@ public static synchronized Map<String, List<String>> getMinimalHeaders() {
...
+            // Remove v from the id
+            return url.substring(url.lastIndexOf('/') + 2);
```
Comment (evermind-zz) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/rumble/RumbleParsingHelper.java]: why not using the matching group (you've specified) instead?

Comment (Benau) [reply extractor/src/main/java/org/schabi/newpipe/extractor/services/rumble/RumbleParsingHelper.java]: Because it was coded by AI, now fixed

**Pull request merged**